relay-metrics/src/aggregator.rs

```diff
@@ -459,6 +460,27 @@ impl fmt::Debug for CostTracker {
     }
 }
 
+/// The minimum trait requirements an [`Aggregator`] can work with.
+///
+/// The aggregator accepts buckets with this type and also returns
+/// aggregated with the same type again.
+///
+/// The aggregator being generic over the contained buckets allows
+/// type safety when aggregating buckets in different states.
+pub trait AggregatorBuckets: Sized + IntoIterator<Item = Bucket> {
```

Comment:
Could the Aggregator, instead of having this trait, simply be marked by the bucket state? IIUC the aggregator always outputs the same bucket type as it gets as input, so for example an `Aggregator<Filtered>` would be an aggregator that aggregates buckets that have already been filtered.

Reply:
Yes, but that requires changing not just the pre aggregation but all aggregation and moving `Buckets` into `relay-metrics`. I don't mind doing it, I didn't to keep the scope smaller.